shared/analytics_tracking/marketo.py

```diff
@@ -0,0 +1,71 @@
+import httpx
+
+from shared.analytics_tracking.base import BaseAnalyticsTool
+from shared.analytics_tracking.events import Event
+from shared.config import get_config
+
+
+class MarketoError(Exception):
+    def __init__(self, error):
+        self.code = error["code"]
+        self.message = error["message"]
+
+    def __str__(self):
+        return f"MarketoError: {self.code} - {self.message}"
+
+
+class Marketo(BaseAnalyticsTool):
+    OAUTH_URL = "/identity/oauth/token"
+    LEAD_URL = "/rest/v1/leads.json"
+
+    def __init__(self) -> None:
+        self.client_id = get_config("setup", "marketo", "client_id", default=None)
+        self.client_secret = get_config(
+            "setup", "marketo", "client_secret", default=None
+        )
+        self.base_url = get_config("setup", "marketo", "base_url", default=None)
+        self.client = httpx.Client()
+
+    @property
+    def token(self):
+        resp = self.retrieve_token()
+        return resp["access_token"]
+
+    @classmethod
+    def is_enabled(cls):
+        return bool(get_config("setup", "marketo", "enabled", default=False))
+
+    def track_event(self, event: Event, *, is_enterprise, context: None):
+        body = {
+            "input": [event.serialize()],
+        }
+        return self.make_rest_request(self.LEAD_URL, method="POST", json=body)
+
+    def make_request(self, url, *args, **kwargs):
+        full_url = self.base_url + url
+        method = kwargs.pop("method", "GET")
+        res = self.client.request(method, full_url, **kwargs)
```

Comment:
The one observation I'd like to make is to maybe add a `try` `except https.HTTPError` around the `self.client.request` method in case the _request_ fails.

And like just emit a warning or something. To prevent this from blowing up.
But chances of this happening are pretty low I suppose...

Reply:
we already catch any exceptions that might happen in manager.py see [here](https://github.com/codecov/shared/blob/main/shared/analytics_tracking/manager.py#L34-L40). I'll add a catch If you think it's worth adding it here too